plugins/utils/flash.nix

```diff
@@ -283,110 +281,80 @@ in
-                      `left` and `right` are always left and right.
-                    '';
-              }
-              ''
-                {
-                  enabled = true;
-                  /* dynamic configuration for ftFT motions */
-                  config = '''
-                    function(opts)
-                      -- autohide flash when in operator-pending mode
-                      opts.autohide = vim.fn.mode(true):find("no") and vim.v.operator == "y"
-
-                      -- disable jump labels when not enabled, when using a count,
-                      -- or when recording/executing registers
-                      opts.jump_labels = opts.jump_labels
-                        and vim.v.count == 0
-                        and vim.fn.reg_executing() == ""
-                        and vim.fn.reg_recording() == ""
-
-                      -- Show jump labels only in operator-pending mode
-                      -- opts.jump_labels = vim.v.count == 0 and vim.fn.mode(true):find("o")
-                    end
-                  ''';
-                  autohide = false;
-                  jumpLabels = false;
-                  multiLine = false;
-                  label = { exclude = "hjkliardc"; };
-                  keys = helpers.listToUnkeyedAttrs [ "f" "F" "t" "T" ";" "," ];
-                  charActions = '''
+            };
+            default = {
+              enabled = true;
+              highlight = {
+                backdrop = false;
+              };
+              jump = {
+                history = true;
+                register = true;
+                nohlsearch = true;
+              };
+              /*
+                forward will be automatically set to the search direction
+                mode is always set to 'search'
+                incremental is set to 'true' when 'incsearch' is enabled
+              */
+              search.automatic = true;
+            };
+          };
+          char = mkModeConfig {
+            description = "options used when flash is activated through a regular search with `/` or `?`";
+            extra = {
+              enabled = helpers.defaultNullOpts.mkBool true "";
+
+              autohide = helpers.defaultNullOpts.mkBool false ''
+                hide after jump when not using jump labels
+              '';
+
+              jumpLabels = helpers.defaultNullOpts.mkBool false "show jump labels";
+
+              multiLine = helpers.defaultNullOpts.mkBool true ''
+                set to `false` to use the current line only
+              '';
+
+              keys =
+                helpers.defaultNullOpts.mkAttrsOf types.str
+                  # FIXME can't show helper func in docs
+                  (helpers.listToUnkeyedAttrs [
+                    "f"
+                    "F"
+                    "t"
+                    "T"
+                    ";"
+                    ","
+                  ])
```

Comment:
```nix
(helpers.listToUnkeyedAttrs (stringToCharacters "fFtT;,"))
```

Or better yet, render the actual helper function as a literal:
```nix
(literalExpression ''helpers.listToUnkeyedAttrs [ "f" "F" "t" "T" ";" "," ]'')
```

Reply:
Honestly, we shouldn't care too much about this.
Lists span over 30 lines ? Fine, let it be.

Reply:
Question is do we want the default to render with `helpers.listToUnkeyedAttrs` or as a bunch of `__unkeyed.` attrs?

Reply:
`(helpers.listToUnkeyedAttrs (stringToCharacters "fFtT;,"))` is fine I guess.

Reply:
Moved to #1714